// sky-tower/dist/output_file.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var reqwest = require('reqwest');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var reqwest__default = /*#__PURE__*/_interopDefaultLegacy(reqwest);

let pid = '';
let uid = '';

const getCurrentPid = () => {
  return pid;
};

const getCurrentUid = () => {
  return uid;
};

const setPid = (p) => {
  pid = p;
};

const setUid = (u) => {
  uid = u;
};

const init = (obj) => {
  setPid(obj.pid);
  setUid(obj.uid);
};

const emitActionEvent = (obj) => {
  reqwest__default['default']({
    url: "http://localhost:8765/emit/action_event",
    method: 'POST',
    withCredentials: false, /* 值为false，表示前端向服务端发请求时不带cookie */
    data: {
      type: 'action',
      time: new Date().getTime(),
      pid: getCurrentPid(),
      uid: getCurrentUid(),
      ...obj
    }
  });
};

const emitCountEvent = (event) => {
  reqwest__default['default']({
    url: "http://localhost:8765/emit/count_event",
    method: 'POST',
    withCredentials: false, /* 值为false，表示前端向服务端发请求时不带cookie */
    data: {
      event,
      type: 'count',
      time: new Date().getTime(),
      pid: getCurrentPid(),
      uid: getCurrentUid()
    }
  });
};

const emitReqEvent = (obj) => {
  reqwest__default['default']({
    url: "http://localhost:8765/emit/req_event",
    method: 'POST',
    withCredentials: false, /* 值为false，表示前端向服务端发请求时不带cookie */
    data: {
      type: 'req',
      time: new Date().getTime(),
      pid: getCurrentPid(),
      uid: getCurrentUid(),
      ...obj
    }
  });
};

const emitRespEvent = (obj) => {
  reqwest__default['default']({
    url: "http://localhost:8765/emit/resp_event",
    method: 'POST',
    withCredentials: false, /* 值为false，表示前端向服务端发请求时不带cookie */
    data: {
      type: 'resp',
      time: new Date().getTime(),
      pid: getCurrentPid(),
      uid: getCurrentUid(),
      ...obj
    }
  });
};

const emitter = {
  emitActionEvent,
  emitCountEvent,
  emitReqEvent,
  emitRespEvent
};

exports.emitter = emitter;
exports.init = init;
